fetch/src/main/java/com/indix/gocd/s3fetch/FetchTask.java

```diff
@@ -68,23 +81,48 @@ private GoPluginApiResponse handleTaskView() {
         return createResponse(responseCode, view);
     }
 
-    private GoPluginApiResponse handleValidation() {
+    private GoPluginApiResponse handleValidation(GoPluginApiRequest request) {
+        Map configMap = (Map) new GsonBuilder().create().fromJson(request.requestBody(), Object.class);
+        Config config = new Config(configMap);
+        FetchExecutor executor = getFetchExecutor(config);
+
         final HashMap validationResult = new HashMap();
+        Map<String, String> errors = executor.validate(config);
+        if (!errors.isEmpty()) {
+            validationResult.put("errors", errors);
+        }
+
         return createResponse(DefaultGoPluginApiResponse.SUCCESS_RESPONSE_CODE, validationResult);
     }
 
     private GoPluginApiResponse handleGetConfigRequest() {
         HashMap config = new HashMap();
+
+        HashMap materialType = new HashMap();
+        materialType.put("default-value", "");
+        materialType.put("required", true);
+        config.put(Constants.MATERIAL_TYPE, materialType);
+
         HashMap repo = new HashMap();
         repo.put("default-value", "");
-        repo.put("required", true);
+        repo.put("required", false);
         config.put(Constants.REPO, repo);
 
         HashMap pkg = new HashMap();
         pkg.put("default-value", "");
-        pkg.put("required", true);
+        pkg.put("required", false);
         config.put(Constants.PACKAGE, pkg);
 
+        HashMap material = new HashMap();
+        material.put("default-value", "");
+        material.put("required", false);
+        config.put(Constants.MATERIAL, material);
+
+        HashMap job = new HashMap();
+        job.put("default-value", "");
+        job.put("required", false);
+        config.put(Constants.JOB, job);
+
         HashMap destination = new HashMap();
         destination.put("default-value", "");
         destination.put("required", true);
```

Comment:
@manojlds Out of curiosity, is there a reason the destination folder is required? We have some pipelines that fetch the artifacts to the workspace root, so we'd need to refactor them.